addons/auth_signup/models/res_users.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import random
+import werkzeug
+
+from ast import literal_eval
+from datetime import datetime, timedelta
+
+from openerp import api, fields, models, _
+from openerp.addons.base.ir.ir_mail_server import MailDeliveryException
+
+from openerp.exceptions import UserError
+from openerp.tools.misc import DEFAULT_SERVER_DATETIME_FORMAT, ustr
+
+
+class SignupError(Exception):
+    pass
+
+def random_token():
+    # the token has an entropy of about 120 bits (6 bits/char * 20 chars)
+    chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'
+    return ''.join(random.SystemRandom().choice(chars) for i in xrange(20))
+
+def now(**kwargs):
+    dt = datetime.now() + timedelta(**kwargs)
+    return dt.strftime(DEFAULT_SERVER_DATETIME_FORMAT)
+
+class ResUsers(models.Model):
+    _inherit = 'res.users'
+
+    @api.depends('login_date')
+    def _get_state(self):
```

Comment:
- depends not required
- prefix it with _compute*
- define it after fields definition